src/state-summary/state-card-input_number.html

```diff
@@ -86,21 +120,37 @@
       value: {
         type: Number,
       },
+      mode: {
+        type: String,
+      }
     };
   }
 
+  hiddenState(sliderwidth) {
+    if (this.mode !== 'slider') return;
+    if (isNaN(sliderwidth)) {
+      sliderwidth = this.$.slider.offsetWidth;
+    }
+    if (sliderwidth < 100) {
+      this.$.sliderstate.hidden = true;
+    } else if (sliderwidth >= 145) {
+      this.$.sliderstate.hidden = false;
+    }
+  }
+
   stateObjectChanged(newVal) {
-    this.min = Number(newVal.attributes.min);
-    this.max = Number(newVal.attributes.max);
-    this.step = Number(newVal.attributes.step);
-    this.value = Number(newVal.state);
-    this.maxlength = this.max.legnth;
-    if (newVal.attributes.mode === 'slider') {
-      this.hiddenbox = true;
-      this.hiddenslider = false;
-    } else {
-      this.hiddenbox = false;
-      this.hiddenslider = true;
+    this.setProperties({
+      min: Number(newVal.attributes.min),
+      max: Number(newVal.attributes.max),
+      step: Number(newVal.attributes.step),
+      value: Number(newVal.state),
+      mode: String(newVal.attributes.mode),
+      maxlength: String(newVal.attributes.max).length,
+      hiddenbox: (newVal.attributes.mode !== 'box'),
+      hiddenslider: (newVal.attributes.mode !== 'slider'),
+    });
+    if (this.mode !== newVal.mode) {
```

Comment:
Did you mean `newVal.attributes.mode`?
In any case this will never be true because `this.mode` is already set by setProperty

Reply:
Yes of course ! This is a mistake
I think I can always call `this.hiddenState()` after a state change anyway, it's not important to check if mode has changed. If it's not slider it will return false. If it is, it will juste recalculate the need to hide the state value.
If there would be a rare case where value changes too often maybe we could check for a `mode` property update, but I don't think that's a use case for an input number.

Reply:
I fixed it in the latest commit: https://github.com/home-assistant/home-assistant-polymer/pull/808/commits/b4eeef2dc33ab4ec89b01b7046e59007a37588de.
Finally decided to go the way of checking for the mode change to be certain to cover every use cases.